ADAL/src/cache/ADTokenCacheAccessor.m

```diff
@@ -191,13 +217,17 @@ - (void)updateCacheToResult:(ADAuthenticationResult *)result
     [self removeItemFromCache:cacheItem
                  refreshToken:refreshToken
                 correlationId:correlationId
+           telemetryRequestId:telemetryRequestId
                         error:result.error];
 }
 
 - (void)updateCacheToItem:(ADTokenCacheItem *)cacheItem
                      MRRT:(BOOL)isMRRT
             correlationId:(NSUUID *)correlationId
+       telemetryRequestId:(NSString*)telemetryRequestId
 {
+    [[ADTelemetry getInstance] startEvent:telemetryRequestId eventName:@"token_cache_write"];
```

Comment:
please change "getInstance" to something else, like "sharedTelemetry" seeing the word "get" in ObjC code looks non-objc.